corehq/apps/users/models.py

```diff
@@ -286,6 +286,23 @@ def permission_names(cls):
             if isinstance(value, BooleanProperty)
         }
 
+    @classmethod
+    def diff(cls, left, right):
+        left_dict = {info.name: info.allow for info in left.to_list()}
+        right_dict = {info.name: info.allow for info in right.to_list()}
+
+        all_names = set(left_dict.keys()) | right_dict.keys()
```

Comment:
Extreme nitpick: the `set()` call is unnecessary. Union of two `dict.keys()` returns a set.